pkg/util/modules/modules.go

```diff
@@ -67,41 +75,46 @@ func (m *Manager) AddDependency(name string, dependsOn ...string) error {
 // InitModuleServices initialises the target module by initialising all its dependencies
 // in the right order. Modules are wrapped in such a way that they start after their
 // dependencies have been started and stop before their dependencies are stopped.
-func (m *Manager) InitModuleServices(target string) (map[string]services.Service, error) {
-	if _, ok := m.modules[target]; !ok {
-		return nil, fmt.Errorf("unrecognised module name: %s", target)
+func (m *Manager) InitModuleServices(name string) error {
+	if !m.IsModuleRegistered(name) {
+		return fmt.Errorf("unrecognised module name: %s", name)
 	}
 
-	servicesMap := map[string]services.Service{}
-
 	// initialize all of our dependencies first
-	deps := m.orderedDeps(target)
-	deps = append(deps, target) // lastly, initialize the requested module
+	deps := m.orderedDeps(name)
+	deps = append(deps, name) // lastly, initialize the requested module
 
 	for ix, n := range deps {
+		// Skip already loaded modules
+		if m.modulesLoaded[n] {
```

Comment:
Couldn't we just check `m.servicesMap`?

Reply:
That wouldn't work... not all modules have a service.